Source/WTF/wtf/haiku/RunLoopHaiku.cpp

```diff
@@ -62,6 +76,42 @@ RunLoop::RunLoop()
 	: m_looper(nullptr)
 {
     m_handler = new LoopHandler();
+
+    // We want to be able to queue messages before the RunLoop::run has been
+    // called, so we need to attach our handler to a looper now. However, we
+    // don't need to start the looper yet.
+
+    BLooper* looper;
+
+    BLooper* currentLooper = BLooper::LooperForThread(find_thread(NULL));
+    if (currentLooper) {
+        // This thread already has a looper (likely the BApplication looper).
+        // Attach our handler to it.
+        looper = currentLooper;
+    } else {
+        // I'm not sure how this code would behave if you created a team without
+        // a BApplication. Comment this ASSERT out and give it a try if you
+        // desire.
+        ASSERT(be_app);
```

Comment:
It's fine for this not to be there in release mode. It's meant to guard against people, say, refactoring the code such that BApplication is not present anymore. In that case, I don't know how the code would behave.

Reply:
it will crash (since be_app is used just below). So I would call debugger instead:

    if (be_app == NULL)
        debugger("Trying to create a runloop without first creating a BApplication");

This makes it start debugger with an helpful error message even in releases. Instead of just crashing with a NULL pointer dereference which can be harder to investigate.